packages/components/src/ClickableStyle/ClickableStyle.tsx

```diff
@@ -23,6 +23,33 @@ export type ClickableStyleProps<IComponent extends React.ElementType> = {
    * The style of the element.
    */
   variant: "flat" | "outline" | "link";
+  /**
+   * Optional icon that can be positioned to the left or right of the other button contents.
+   *
+   * Default position is 'left'. Use the iconPlacement prop to move it to the right side.
+   */
+  icon?: {
```

Comment:
Could it make sense to split these, so there's an `icon` prop that just takes the element, then `iconProps` that has `placement` and `horizontalMargin` fields? Thinking that most of the time, people won't need to pass the extra props